src/main/java/org/testng/internal/TestListenerHelper.java

```diff
@@ -19,31 +20,60 @@ private TestListenerHelper() {
     // Utility class. Defeat instantiation.
   }
 
-  static void runPreConfigurationListeners(ITestResult tr, List<IConfigurationListener> listeners) {
+  static void runPreConfigurationListeners(ITestResult tr, ITestNGMethod tm, List<IConfigurationListener> listeners) {
     for (IConfigurationListener icl : listeners) {
       icl.beforeConfiguration(tr);
+      try {
+        icl.beforeConfiguration(tr, tm);
```

Comment:
By default, methods with `tm` should call methods without `tm`, then no need to call both of them.

Reply:
Same explanation as for the other comment. This chaining via `default` methods will cause the one arg method to not get invoked when a user implements both the methods in the interface.